src/BigtableClient.php

```diff
@@ -151,6 +162,12 @@ public function __construct(array $config = [])
      */
     public function table($instanceId, $tableId, array $options = [])
     {
+        if ($this->pingAndWarm) {
+            $this->gapicClient->pingAndWarm(
```

Comment:
whats the default deadline here?
Also, would it be possible to add a simple lru cache to avoid re-warming instances that have been warmed already?

Reply:
The default deadline is defined in the JSON config. It would need to be specifically overwritten by the customer to change, so I think we're okay here.

https://github.com/googleapis/google-cloud-php-bigtable/blob/eb55e2364a86ab337383ff0c118c51fa32b11f65/src/V2/resources/bigtable_client_config.json#L42-L46

As for a LRU cache - good idea! I've added a `pingAndWarmCalled` array, with `$instanceId` as a key, to determine if it's been called. Do you think this will suffice? See a3aebae.

Reply:
re deadline. Can you the github comment as a code comment in the file just so take when someone is looking at it in the future they dont have to reverse engineer like we jsut did?

ie something like:
// The default deadline is configured by the service config (src/V2/resources/bigtable_client_config.json). By default the deadline should be high enough to absorb cold connection latencies.

re - cache
Yea, I like the simplicity. We might revisit in the future if we decide to mainline this

Reply:
Good idea! Comment added in ef59a3a

Reply:
As for `pingAndWarmCalled`, we could make that a static variable, and use `GapicClient::instanceName` as the key. This would ensure that even creating another `BigtableClient` instance wouldn't call it.

Reply:
I dont think you want it as a static variable as you dont know if the 2 client instances are using the same set of credentials.

Reply:
I dont think you want it as a static variable as you dont know if the 2 client instances are using the same set of credentials. So might end up using different persistent channels

Reply:
good point! reverted